src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -116,7 +116,11 @@ private MethodHandles() { }  // do not instantiate
     @CallerSensitive
     @ForceInline // to ensure Reflection.getCallerClass optimization
     public static Lookup lookup() {
-        return new Lookup(Reflection.getCallerClass());
+        Class<?> c = Reflection.getCallerClass();
+        if (c == null) {
+            throw new IllegalCallerException();
```

Comment:
```suggestion
            throw new IllegalCallerException("no caller frame");
```

Reply:
The javadoc needs to be updated to specify this `IllegalCallerException` be thrown.

```
    * @throws IllegalCallerException if there is no caller frame on the stack when called
    * directly from a JNI attached thread
```